### PR TITLE
chore(weave): Add materialized views that will make it easier to do billing calculations

### PR DESCRIPTION
**Pull request by adrnswanberg**
## Description

Add materialized views that provide usage stats for our major tables.

## Testing

How was this PR tested?
It hasn't, yet :)

### CODE REVIEW

File: weave/trace_server/migrations/007_stats_views.down.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+DROP TABLE IF EXISTS weave_trace_db.calls_merged_stats;
```
Comment (tssweeney) [inline weave/trace_server/migrations/007_stats_views.down.sql:1]: calls merged is one of the few that we don't need actually

Comment (adrnswanberg) [reply weave/trace_server/migrations/007_stats_views.down.sql:1]: still helpful to make counting up usage dramatically faster

Comment (tssweeney) [reply weave/trace_server/migrations/007_stats_views.down.sql:1]: fair enough - any reason you did not do the other tables?

File: weave/trace_server/migrations/007_stats_views.down.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+DROP VIEW IF EXISTS weave_trace_db.calls_merged_stats_view;
+
+DROP TABLE IF EXISTS weave_trace_db.object_versions_stats;
+DROP VIEW IF EXISTS weave_trace_db.object_versions_stats_view;
+
+DROP TABLE IF EXISTS weave_trace_db.table_rows_stats;
+DROP VIEW IF EXISTS weave_trace_db.table_rows_stats_view;
+
+DROP TABLE IF EXISTS weave_trace_db.files_stats;
+DROP VIEW IF EXISTS weave_trace_db.files_stats_view;
```
Comment (tssweeney) [inline weave/trace_server/migrations/007_stats_views.down.sql:11]: Note: you don't actually need the `weave_trace_db` prefix here, or anywhere as the DB is a param of the entire migration process

File: weave/trace_server/migrations/007_stats_views.up.sql [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+(
+    project_id String,
+    id String,
+    trace_id SimpleAggregateFunction(any, Nullable(String)),
+    parent_id SimpleAggregateFunction(any, Nullable(String)),
+    op_name SimpleAggregateFunction(any, Nullable(String)),
+    started_at SimpleAggregateFunction(any, Nullable(DateTime64(3))),
+    attributes_size_bytes SimpleAggregateFunction(any, Nullable(UInt64)),
+    inputs_size_bytes SimpleAggregateFunction(any, Nullable(UInt64)),
+    input_refs SimpleAggregateFunction(groupArrayArray, Array(String)),
```
Comment (tssweeney) [inline weave/trace_server/migrations/007_stats_views.up.sql:99]: not sure if you wanted to trim down the fields, but you could probably drop many of these if we just want the size stats.

File: weave/trace_server/migrations/007_stats_views.up.sql [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+CREATE MATERIALIZED VIEW object_versions_stats_view
+TO weave_trace_db.object_versions_stats
+AS
+SELECT
+    project_id,
+    kind,
+    object_id,
+    digest,
+    anySimpleState(base_object_class) AS base_object_class,
+    anySimpleState(refs) AS refs,
```
Comment (tssweeney) [inline weave/trace_server/migrations/007_stats_views.up.sql:81]: similar here, do we really need these other fields?

File: weave/trace_server/migrations/007_stats_views.down.sql (hunk 1 above)
Comment (adrnswanberg) [reply weave/trace_server/migrations/007_stats_views.down.sql:1]: you mean like tables? we only need this for tables we bill on. and besides feedback, no other tables have user-defined fields of arbitrary length

Comment (tssweeney) [reply weave/trace_server/migrations/007_stats_views.down.sql:1]: feedback does, but i guess we don't bill that right now

Comment (adrnswanberg) [reply weave/trace_server/migrations/007_stats_views.down.sql]: added one for feedback

File: weave/trace_server/migrations/007_stats_views.down.sql (hunk 2 above)
Comment (adrnswanberg) [reply weave/trace_server/migrations/007_stats_views.down.sql]: fixed

File: weave/trace_server/migrations/007_stats_views.up.sql (hunk 3 above)
Comment (adrnswanberg) [reply weave/trace_server/migrations/007_stats_views.up.sql:130]: felt easier to include them..they might come in handy for quick analysis? we can always drop them later. easier to drop than include later as backfills are hard.

**Comment by adrnswanberg** [comment]
@tssweeney Finally got around to testing this -- all working ok.

**Review by tssweeney — APPROVED** [review]

**Review by tssweeney — CHANGES_REQUESTED** [review]
Please bump the migration number (requesting changes to make sure we don't forget)

**Pull request merged**